crates/ruff/src/rules/flake8_simplify/rules/ast_ifexp.rs

```diff
@@ -30,21 +31,29 @@ use crate::registry::AsRule;
 /// - [Python documentation: Truth Value Testing](https://docs.python.org/3/library/stdtypes.html#truth-value-testing)
 #[violation]
 pub struct IfExprWithTrueFalse {
-    expr: String,
+    is_compare: bool,
 }
 
 impl Violation for IfExprWithTrueFalse {
     const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
 
     #[derive_message_formats]
     fn message(&self) -> String {
-        let IfExprWithTrueFalse { expr } = self;
-        format!("Use `bool({expr})` instead of `True if {expr} else False`")
+        let IfExprWithTrueFalse { is_compare } = self;
+        if !is_compare {
+            format!("Use `bool(...)` instead of `True if ... else False`")
+        } else {
+            format!("Remove unnecessary `True if ... else False` from comparison")
+        }
```

Comment:
I decided to remove the raw source code from these. I think we're hoping that this information is instead conveyed by showing code frames.